.github/workflows/collection-continuous-integration.yml

```diff
@@ -0,0 +1,258 @@
+    - name: Build a collection tarball
+      run: >-
+        ~/.local/bin/ansible-galaxy collection build --output-path
+        "${GITHUB_WORKSPACE}/.cache/collection-tarballs"
+    - name: Store migrated collection artifacts
+      uses: actions/upload-artifact@v1
+      with:
+        name: >-
+          collection
+        path: .cache/collection-tarballs
+
+  sanity-test-collection-via-vms:
+    name: Sanity in VM ${{ matrix.os.vm || 'ubuntu-latest' }}
+    needs:
+    - build-collection-artifact
+    runs-on: ${{ matrix.os.vm || 'ubuntu-latest' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        ansible-version:
+          - git+https://github.com/ansible/ansible.git@devel
+        os:
+          - vm: ubuntu-16.04
+          - vm: macos-latest
+        python-version:
+          - 3.7
+          - 2.7
+
+    steps:
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Set up pip cache
+      uses: actions/cache@v1
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip-${{ github.ref }}-sanity-VMs
+        restore-keys: |
+          ${{ runner.os }}-pip-
+          ${{ runner.os }}-
+    - name: Install Ansible ${{ matrix.ansible-version }}
+      run: >-
+        python -m pip install --user ${{ matrix.ansible-version }} virtualenv
+    - name: Download migrated collection artifacts
+      uses: actions/download-artifact@v1
+      with:
+        name: >-
+          collection
+        path: .cache/collection-tarballs
+    - name: Install the collection tarball
+      run: >-
+        ~/.local/bin/ansible-galaxy collection install .cache/collection-tarballs/*.tar.gz
+    - name: Run collection sanity tests
+      run: >-
+        ~/.local/bin/ansible-test sanity --color --requirements --venv --python "${{ matrix.python-version }}" -vvv
+      working-directory: >-
+        /${{ runner.os == 'Linux' && 'home' || 'Users' }}/runner/.ansible/collections/ansible_collections/containers/podman
+
+  sanity-test-collection-via-containers:
+    name: Sanity in container via Python ${{ matrix.python-version }}
+    needs:
+    - build-collection-artifact
+    runs-on: ${{ matrix.runner-os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        runner-os:
+          - ubuntu-latest
+        runner-python-version:
+          - 3.7
+        ansible-version:
+          - ansible==2.9.5
+          - git+https://github.com/ansible/ansible.git@devel
+        python-version:
+          - 2.7
+          - 3.7
+
+    steps:
+    - name: Set up Python ${{ matrix.runner-python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.runner-python-version }}
+    - name: Set up pip cache
+      uses: actions/cache@v1
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip-${{ github.ref }}-sanity-containers
```

Comment:
For best results, you should also bake-in the deps files hashes

Reply:
yeah, good idea, will do overall optimization in followups